kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,838 @@
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "snake_sbm_utilities.h"
+
+namespace Kratos
+{
+
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(ModelPart& rIgaModelPart,
+                                                    ModelPart& rSkinModelPartInnerInitial,
+                                                    ModelPart& rSkinModelPartOuterInitial, 
+                                                    ModelPart& rSkinModelPart, 
+                                                    int rEchoLevel, 
+                                                    Vector& knotVectorU, 
+                                                    Vector& knotVectorV,
+                                                    const Parameters mParameters) { 
+        
+        // Check
+        KRATOS_ERROR_IF_NOT(mParameters.Has("sbm_parameters")) << "sbm_parameters has not been defined in the nurbs modeler" << std::endl;
+        
+        // Initilize the property of skin_model_part_in and out
+        if (rSkinModelPartInnerInitial.Nodes().size()>0) {
+        
+            CreateTheSnakeCoordinates(rIgaModelPart, rSkinModelPartInnerInitial, rSkinModelPart, rEchoLevel, knotVectorU, knotVectorV, mParameters, true);
+                
+        }
+        if (rSkinModelPartOuterInitial.Nodes().size()>0) {
+
+            CreateTheSnakeCoordinates(rIgaModelPart, rSkinModelPartOuterInitial, rSkinModelPart, rEchoLevel, knotVectorU, knotVectorV, mParameters, false);
+        }
+    }   
+
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(ModelPart& rIgaModelPart, 
+                                                      ModelPart& rSkinModelPartInitial, 
+                                                      ModelPart& rSkinModelPart,
+                                                      int rEchoLevel, 
+                                                      Vector& knotVectorU, 
+                                                      Vector& knotVectorV,
+                                                      const Parameters mParameters,
+                                                      bool isInner) 
+    { 
+        
+        std::string surrogate_sub_model_part_name; 
+        std::string skin_sub_model_part_name; 
+        // ModelPart skin_sub_model_part; 
+        if (isInner) {
+            surrogate_sub_model_part_name = "surrogate_inner";
+            skin_sub_model_part_name = "inner";
+        }
+        else {
+            surrogate_sub_model_part_name = "surrogate_outer";
+            skin_sub_model_part_name = "outer";
+        }
+        
+        ModelPart& skin_sub_model_part = rSkinModelPart.GetSubModelPart(skin_sub_model_part_name);
+        ModelPart& surrogate_sub_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
```

Comment:
```suggestion
        auto& r_skin_sub_model_part = rSkinModelPart.GetSubModelPart(skin_sub_model_part_name);
        auto& r_surrogate_sub_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
```
Note the `auto` return type and the `r_` following our convention to indicate that the local var is a reference.